torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -345,4 +345,68 @@ Tensor & VariableType::detach_(Tensor & self) const {
   return self;
 }
 
+Tensor VariableType::_sparse_coo_tensor_with_dims_and_tensors(int64_t sparse_dim, int64_t dense_dim, IntArrayRef size, const Tensor & indices, const Tensor & values, const TensorOptions & options) const {
+  RECORD_FUNCTION("_sparse_coo_tensor_with_dims_and_tensors", std::vector<c10::IValue>({indices, values}), Function::peek_at_next_sequence_nr());
+  auto indices_ = Tensor(indices.unsafeGetTensorImpl()->shallow_copy_and_detach());
+  auto values_ = Tensor(values.unsafeGetTensorImpl()->shallow_copy_and_detach());
```

Comment:
The only change in the implementation of `VariableType::_sparse_coo_tensor_with_dims_and_tensors` is these two lines for `indices_` and `values_` (they used to be `auto& indices_ = unpack(indices, "indices", 3)` and `auto& values_ = unpack(values, "values", 4)`). We need this manual implementation because of the following reason:

SparseTensorImpl's `set_indices_and_values_unsafe(indices, values)` function expects `indices` and `values` to not requires grad. Before this patch, we use `unpack()` to unwrap the Variable and obtain the underlying data tensor, which is guaranteed to not requires grad. After this patch, `unpack()` doesn't do the unwrapping anymore, and we need a way to create non-requires-grad `indices_` and `values_` tensors that contains the same data as `indices` and `values`, which is achieved by shallow-copying `indices` and `values` and wrapping them in new Tensors here.